### PR TITLE
Bump static analysis

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

BC

## Changelog

```markdown
### Fixed
- multiple phpdoc
```

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Does someone knows the `sonata_status_class` twig method ?
https://github.com/search?p=2&q=org%3Asonata-project+sonata_status_class&type=Code

https://github.com/sonata-project/twig-extensions/blob/1.x/src/Extension/StatusRuntime.php#L45
The signature is kinda weird.
I think it's related to https://github.com/sonata-project/twig-extensions/pull/90
Shouldn't we deprecated some authorized value or split the method ?

I feel like passing a string should be deprecated in favor of using `sonata_flashmessages_class`.
Am I right @wbloszyk ?

Shouldn't we also deprecate the `StatusClassRendererInterface` interface if the FlashManager won't implement it in next major ?

**Comment by VincentLanglet** [comment]
> Does someone knows the `sonata_status_class` twig method ? https://github.com/search?p=2&q=org%3Asonata-project+sonata_status_class&type=Code
> 
> https://github.com/sonata-project/twig-extensions/blob/1.x/src/Extension/StatusRuntime.php#L45 The signature is kinda weird. I think it's related to #90 Shouldn't we deprecated some authorized value or split the method ?
> 
> I feel like passing a string should be deprecated in favor of using `sonata_flashmessages_class`. Am I right @wbloszyk ?
> 
> Shouldn't we also deprecate the `StatusClassRendererInterface` interface if the FlashManager won't implement it in next major ?

I need help on this @sonata-project/contributors

**Comment by jordisala1991** [comment]
> > Does someone knows the `sonata_status_class` twig method ? https://github.com/search?p=2&q=org%3Asonata-project+sonata_status_class&type=Code
> 
> > 
> 
> > https://github.com/sonata-project/twig-extensions/blob/1.x/src/Extension/StatusRuntime.php#L45 The signature is kinda weird. I think it's related to #90 Shouldn't we deprecated some authorized value or split the method ?
> 
> > 
> 
> > I feel like passing a string should be deprecated in favor of using `sonata_flashmessages_class`. Am I right @wbloszyk ?
> 
> > 
> 
> > Shouldn't we also deprecate the `StatusClassRendererInterface` interface if the FlashManager won't implement it in next major ?
> 
> 
> 
> I need help on this @sonata-project/contributors 

Is this something that blocks this Pr or can you make an issue an merge this one first?

**Comment by VincentLanglet** [comment]
> Is this something that blocks this Pr or can you make an issue an merge this one first?

It's blocking me for the level 8 of phpstan and 3 of psalm.

**Comment by VincentLanglet** [comment]
The only error level 8 which is blocking me is
```
 ------ ------------------------------------------------------------------------------------------------------------------------------------- 
  Line   src/Extension/StatusRuntime.php                                                                                                      
 ------ ------------------------------------------------------------------------------------------------------------------------------------- 
  47     Parameter #1 $object of method Sonata\Twig\Extension\StatusRuntime::statusClassForFlashManager() expects string, string|null given.  
 ------ ------------------------------------------------------------------------------------------------------------------------------------- 
```

`statusClass()` seems to support lot of syntax different. Too much to me ; we should simplified this.

Usage is https://github.com/search?p=1&q=org%3Asonata-project+sonata_status_class&type=Code
`object|sonata_status_class(string/null, 'danger')` most of the time.

So I'm gonna deprecate other syntaxes.

File: src/FlashMessage/FlashManager.php [hunk 1]
```diff
@@ -25,54 +26,50 @@
...
      *
-     * @deprecated since sonata-project/twig-extensions 1.7. Use $requestStack->getSession() instead.
-     */
-    private $session;
-
-    /**
-     * @var RequestStack|null
+     * @var RequestStack|Session
      */
     private $requestStack;
```
Comment (core23) [inline src/FlashMessage/FlashManager.php]:
```suggestion
    private $requestStackOrSession;
```

Comment (VincentLanglet) [reply src/FlashMessage/FlashManager.php]: Done

**Review by core23 — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**